master/buildbot/data/buildrequests.py

```diff
@@ -143,8 +143,10 @@ def get(self, resultSpec, kwargs):
             claimed=claimed,
             bsid=bsid,
             resultSpec=resultSpec)
-        defer.returnValue(
-            [(yield self.db2data(br)) for br in buildrequests])
+        results = []
+        for br in buildrequests:
+            results.append((yield self.db2data(br)))
+        defer.returnValue(results)
```

Comment:
I'm not well versed in Twisted, but could this be replaced with a `DeferredList`?